### DISCUSSION TITLE
gaps in hybrid RNA-seq assembly

### DISCUSSION THREAD

**Discussion by glopatriello** [ablab/spades#861] [Q&A]
Hi there!

I have used rnaSPAdes to perform a de novo transcriptome assembly using both Illumina and ONT sequencing data. In my results, I found that 1% of my transcripts have gaps into their sequence. I was wondering why the software has introduced these gaps.

Best regards,
Giulia

**Comment by andrewprzh** [comment]
Hi Giulia,

This is completely normal. Such sequences appear when low-abundant transcripts are not fully sequences, but their fragments can be connected by paired-reads, for example.

Best
Adnrey

**Comment by glopatriello** [comment]
Dear Andrey,

Thank you for your response. I was wondering based on which criteria rnaSPAdes decides the length of the gap to introduce in the assembly. Furthermore, I would like to ask how you recommend to resolve the gaps.

Best,
Giulia

**Reply by andrewprzh** [reply]
Dear Giulia

There could be 2 types of gaps in the assembly:
1. coverage gaps - in this case the length of the gap is estimated using average insert size of paired reads;
2. unresolved tandem repeats - always 100 bp gaps, but extremely rare for transcriptome assembly.

Unfortunately I'm not really aware, but I presume there could be tools for polishing genome assembly.
However, if the assembler inserted a gap - I doubt it can be fixed without additional sequencing. You can try mapping reads onto the assembly and check whether any reads cover the gap. If yes -- try polishing tools.

Best
Andrey
